assets/less/sidebar.less

```diff
@@ -131,31 +131,75 @@
   }
 
   .sidebar-search {
-    margin: 18px 30px;
-    .display(flex);
+    margin-top: 12px;
 
     &.selected .search-button,
     .search-button:hover {
       color: @purple;
+      opacity: 1;
+    }
+
+    .search-label {
+      position: relative;
+      width: 100%;
     }
 
     .search-button {
       font-size: 14px;
       color: @lightestGray;
       background-color: transparent;
       border: none;
-      padding: 2px 0 0 0;
+      cursor: pointer;
+      left: 22px;
       margin: 0;
+      opacity: 0.5;
+      padding: 3px 1px 3px 0;
+      position: absolute;
+      top: 18px;
+      transition: 0.12s all ease-out;
+      z-index: 2;
     }
 
-    .search-input {
+    .search-close-button {
+      font-size: 16px;
+      color: @lightestGray;
       background-color: transparent;
       border: none;
-      border-radius: 0;
-      border-bottom: 1px solid @mediumGray;
+      cursor: pointer;
+      right: 18px;
+      margin: 0;
+      opacity: 0.5;
+      padding: 3px 1px 3px 0;
+      position: absolute;
+      transform: scaleY(0);
+      top: 17px;
+      transition: 0.15s transform ease-out;
+      z-index: 2;
+
+      .icon-cross {
+        font-weight: bold;
+      }
+
+      &:hover {
+        opacity: 0.7;
+      }
+
+      &:focus,
+      &:hover {
+        outline: none;
+      }
+    }
+
+    .search-input {
+      background-color: darken(@darkGray, 8%);
+      border: none;
+      border-radius: 4px;
       color: @lightestGray;
-      margin-left: 5px;
-      height: 20px;
+      margin-left: 12px;
+      margin-right: 10px;
```

Comment:
The most unimportant nitpick on earth: you either don't need the margin (since we have the width) or you should make the margin be 12px so it totals 300px with the width. :D

Reply:
Removed! :D